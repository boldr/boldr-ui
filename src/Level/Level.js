/* @flow */
import React from 'react';
import classNames from 'classnames';
import { StyleClasses } from '../theme/styleClasses';
import { createWrappedComponent } from '../util/boldrui';

export type Props = {
  tag?: string,
  isMobile?: boolean,
  className?: string,
  href?: string,
};

const BASE_ELEMENT = StyleClasses.LEVEL;
export function Level({ tag = 'nav', ...props }: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      'is-mobile': props.isMobile,
    },
    props.className,
  );

  const { ...HTMLProps } = props;

  return React.createElement(tag, { ...HTMLProps, className });
}

export default createWrappedComponent(Level);
