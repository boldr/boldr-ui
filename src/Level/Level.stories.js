import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Button from '../Button';
import { Input, Control, FormField } from '../Form';
import { Level, LevelItem, LevelRight, LevelLeft } from './index';

storiesOf('Level', module).add('default', () => (
  <Level>
    <LevelLeft>
      <LevelItem>
        <strong>123</strong> posts
      </LevelItem>
      <LevelItem>
        <FormField hasAddons>
          <Control>
            <Input placeholder="Find a post" />
          </Control>
          <Control>
            <Button>Search</Button>
          </Control>
        </FormField>
      </LevelItem>
    </LevelLeft>
    <LevelRight>
      <LevelItem>
        <strong>All</strong>
      </LevelItem>
      <LevelItem>
        <a>Published</a>
      </LevelItem>
      <LevelItem>
        <a>Draft</a>
      </LevelItem>
      <LevelItem>
        <a>Deleted</a>
      </LevelItem>
      <LevelItem>
        <Button kind="success">New</Button>
      </LevelItem>
    </LevelRight>
  </Level>
));
