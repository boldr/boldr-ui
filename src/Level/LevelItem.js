/* @flow */
import React from 'react';
import classNames from 'classnames';
import { StyleClasses } from '../theme/styleClasses';
import { createWrappedComponent } from '../util/boldrui';

export type Props = {
  tag?: string,
  isFlexible?: boolean,
  className?: string,
  href?: string,
};
const BASE_ELEMENT = StyleClasses.LEVEL_ITEM;
export function LevelItem({ tag = 'div', ...props }: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      'is-flexible': props.isFlexible,
    },
    props.className,
  );

  const { ...HTMLProps } = props;

  return React.createElement(props.href ? 'a' : tag, { ...HTMLProps, className });
}

export default createWrappedComponent(LevelItem);
