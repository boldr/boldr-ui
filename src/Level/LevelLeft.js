/* @flow */
import React from 'react';
import classNames from 'classnames';
import { StyleClasses } from '../theme/styleClasses';
import { createWrappedComponent } from '../util/boldrui';

export type Props = {
  tag?: string,
  className?: string,
};

const BASE_ELEMENT = StyleClasses.LEVEL_LEFT;

export function LevelLeft({ tag = 'div', ...props }: Props) {
  const className = classNames(BASE_ELEMENT, props.className);

  return React.createElement(tag, { ...props, className });
}

export default createWrappedComponent(LevelLeft);
