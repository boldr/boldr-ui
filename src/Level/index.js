export { default as Level } from './Level';
export { default as LevelItem } from './LevelItem';
export { default as LevelLeft } from './LevelLeft';
export { default as LevelRight } from './LevelRight';
