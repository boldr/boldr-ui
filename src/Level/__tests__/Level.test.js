import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Level } from '../Level';

describe('Level', () => {
  it('should render a nav with .boldrui-level', () => {
    const component = shallow(<Level>My Level</Level>);
    expect(component.contains(<nav className="boldrui-level">My Level</nav>)).toBe(true);
  });

  it('should render a div with .boldrui-level', () => {
    const component = shallow(<Level tag="div">My Level</Level>);
    expect(component.contains(<div className="boldrui-level">My Level</div>)).toBe(true);
  });

  it('should render a nav with .boldrui-level and modifiers', () => {
    const component = shallow(<Level isMobile>My Level</Level>);
    expect(component.hasClass('boldrui-level')).toBe(true);
    expect(component.hasClass('is-mobile')).toBe(true);
  });

  it('should render a nav with .boldrui-level, modifiers and classNames', () => {
    const component = shallow(
      <Level isMobile className="custom">
        My Level
      </Level>,
    );
    expect(component.hasClass('boldrui-level')).toBe(true);
    expect(component.hasClass('is-mobile')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });

  it('should render a nav with .boldrui-level and custom classNames', () => {
    const component = shallow(
      <Level className="custom">
        <span>Any Content</span>
      </Level>,
    );
    expect(component.hasClass('boldrui-level')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<Level>My Level</Level>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
