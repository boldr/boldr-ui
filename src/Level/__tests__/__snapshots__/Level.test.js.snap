// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level should match the snapshot 1`] = `
<nav
  className="boldrui-level"
>
  My Level
</nav>
`;
